bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -235,12 +244,18 @@ public ResultArrayIterator(int arraySize) {
     }
 
     @Override
-    public Result[] next(ResultScanner<Row> resultScanner)
+    public Result[] next(ResultScanner<Row> resultScanner, ByteKeyRangeTracker rangeTracker)
         throws IOException {
       List<Result> results = new ArrayList<>();
       for (int i = 0; i < arraySize; i++) {
         Row row = resultScanner.next();
         if (row == null) {
+          // The scan completed.
+          break;
+        }
+        ByteKey key = ByteKey.copyFrom(row.getKey().toByteArray());
```

Comment:
There's a mismatch between our implementation of ByteString (which is shaded and renamed) and Dataflow's default package ByteString.  That's why the copying happens.  I can create a copy of the ByteKey, ByteKeyRange and ByteKeyRangeTracker classes from Dataflow that use our shaded ByteString.  That's what would be required to remove this copying.